client/state/plugins/wporg/actions.js

```diff
@@ -52,3 +65,144 @@ export function fetchPluginData( pluginSlug ) {
 		}
 	};
 }
+
+/**
+ * Helper thunk for receiving a specific data or retrieve error of plugins list.
+ * Handles plugin list normalization internally.
+ *
+ * @param {string} category   Plugin category
+ * @param {number} page       Page (1-based)
+ * @param {string} searchTerm Search term
+ * @param {Array}  data       List of found plugins, not defined if there was an error
+ * @param {object} error      Error object, undefined if the plugins were fetched successfully
+ * @param {object} pagination Paginatioin data, as retrieved from the API response.
+ * @returns {Function} Action thunk
+ */
+function receivePluginsList( category, page, searchTerm, data, error, pagination = null ) {
+	return ( dispatch ) =>
+		dispatch( {
+			type: PLUGINS_WPORG_LIST_RECEIVE,
+			category,
+			page,
+			searchTerm,
+			data: normalizePluginsList( data ),
+			pagination,
+			error,
+		} );
+}
```

Comment:
Does this need to be a `thunk`? It could be just a regular action, no?

Reply:
Yep, good point. Updated in 09db4c427071fc90423a5f03e46f381cae25333f